### ISSUE TITLE
can't convert nil into Integer (TypeError)

### ISSUE THREAD

**Issue by ethaning** [presidentbeef/brakeman#1672]
### Background

Brakeman version: 5.2.0
Rails version: 6.0.4.4
Ruby version: 2.6.3

### Issue

What problem are you seeing?

When I run ```bundle exec brakeman```, I get the following error: ```can't convert nil into Integer (TypeError)```

#### Other Error

Stack trace:

```
...
 - CheckXMLDoS
 - CheckYAMLParsing
Finding all mass assignments
Unhandled SQL method: payouts
Processing all mass assignments
Unhandled SQL method: filtered_at_date
Unhandled SQL method: claimable
Unhandled SQL method: claimable
Unhandled SQL method: claimable
Unhandled SQL method: filtered_at_date
Processing possible SQL calls
Checks finished, collecting results...
Generating report...
Traceback (most recent call last):
  23: from /usr/local/rvm/gems/ruby-2.6.3@paperless/bin/ruby_executable_hooks:24:in `<main>'
  22: from /usr/local/rvm/gems/ruby-2.6.3@paperless/bin/ruby_executable_hooks:24:in `eval'
  21: from /usr/local/rvm/gems/ruby-2.6.3@paperless/bin/brakeman:23:in `<main>'
  20: from /usr/local/rvm/gems/ruby-2.6.3@paperless/bin/brakeman:23:in `load'
  19: from /usr/local/rvm/gems/ruby-2.6.3@paperless/gems/brakeman-5.2.0/bin/brakeman:10:in `<top (required)>'
  18: from /usr/local/rvm/gems/ruby-2.6.3@paperless/gems/brakeman-5.2.0/lib/brakeman/commandline.rb:20:in `start'
  17: from /usr/local/rvm/gems/ruby-2.6.3@paperless/gems/brakeman-5.2.0/lib/brakeman/commandline.rb:35:in `run'
  16: from /usr/local/rvm/gems/ruby-2.6.3@paperless/gems/brakeman-5.2.0/lib/brakeman/commandline.rb:166:in `run_report'
  15: from /usr/local/rvm/gems/ruby-2.6.3@paperless/gems/brakeman-5.2.0/lib/brakeman/commandline.rb:125:in `regular_report'
  14: from /usr/local/rvm/gems/ruby-2.6.3@paperless/gems/brakeman-5.2.0/lib/brakeman/commandline.rb:157:in `run_brakeman'
  13: from /usr/local/rvm/gems/ruby-2.6.3@paperless/gems/brakeman-5.2.0/lib/brakeman.rb:86:in `run'
  12: from /usr/local/rvm/gems/ruby-2.6.3@paperless/gems/brakeman-5.2.0/lib/brakeman.rb:411:in `scan'
  11: from /usr/local/rvm/gems/ruby-2.6.3@paperless/gems/brakeman-5.2.0/lib/brakeman.rb:447:in `write_report_to_formats'
  10: from /usr/local/rvm/gems/ruby-2.6.3@paperless/gems/brakeman-5.2.0/lib/brakeman/report/pager.rb:15:in `page_report'
   9: from /usr/local/rvm/gems/ruby-2.6.3@paperless/gems/brakeman-5.2.0/lib/brakeman/report.rb:38:in `format'
   8: from /usr/local/rvm/gems/ruby-2.6.3@paperless/gems/brakeman-5.2.0/lib/brakeman/report.rb:95:in `to_text'
   7: from /usr/local/rvm/gems/ruby-2.6.3@paperless/gems/brakeman-5.2.0/lib/brakeman/report.rb:107:in `generate'
   6: from /usr/local/rvm/gems/ruby-2.6.3@paperless/gems/brakeman-5.2.0/lib/brakeman/report/report_text.rb:23:in `generate_report'
   5: from /usr/local/rvm/gems/ruby-2.6.3@paperless/gems/brakeman-5.2.0/lib/brakeman/report/report_text.rb:94:in `generate_warnings'
   4: from /usr/local/rvm/gems/ruby-2.6.3@paperless/gems/brakeman-5.2.0/lib/brakeman/report/report_text.rb:94:in `sort_by'
   3: from /usr/local/rvm/gems/ruby-2.6.3@paperless/gems/brakeman-5.2.0/lib/brakeman/report/report_text.rb:94:in `each'
   2: from /usr/local/rvm/gems/ruby-2.6.3@paperless/gems/brakeman-5.2.0/lib/brakeman/report/report_text.rb:95:in `block in generate_warnings'
   1: from /usr/local/rvm/gems/ruby-2.6.3@paperless/gems/brakeman-5.2.0/lib/brakeman/warning.rb:251:in `fingerprint'
/usr/local/rvm/gems/ruby-2.6.3@paperless/gems/brakeman-5.2.0/lib/brakeman/warning.rb:251:in `sprintf': can't convert nil into Integer (TypeError)

```

Any help appreciated :pray:

**Comment by klazuka** [comment]
I debugged the same problem and reported my findings in #1671

**Comment by presidentbeef** [comment]
Fixed with 5.2.1. Thanks and apologies!
